doc/Database.md

```diff
@@ -0,0 +1,105 @@
+|------------------|----------|--------|-------------|-------------------------------------------------------------------|
+| node_enroll_data | TEXT     | string | PRIMARY KEY | Used as an AA. Valid values: `utxo`, `random_seed`                |
+| val              | TEXT     | Hash   | NOT NULL    | In both case a hash, but the node could just query `ValidatorSet` |
+
+
+### `accumulated_fee` table
+
+Note sure why we need this.
+
+| Field name | SQL Type | D type     | Attributes  | Comment             |
+|------------|----------|------------|-------------|---------------------|
+| public_key | TEXT     | PublicKey  | PRIMARY KEY |                     |
+| fee        | TEXT     | Enrollment |             | Should be `INTEGER` |
+
+## Cache DB
+
+### `tx_pool` table
+
+Stores the pending transaction. Should be made readable.
+
+| Field name | SQL Type | D type      | Attributes  | Comment           |
+|------------|----------|-------------|-------------|-------------------|
+| key        | TEXT     | Hash        | PRIMARY KEY | Transaction hash  |
+| val        | BLOB     | Transaction | NOT NULL    | Binary-serialized |
+
+### `enrollment_pool` table
+
+Store the `Enrollment` that haven't made it to a `Block` yet. Should be made readable.
+
+| Field name   | SQL Type | D type     | Attributes  | Comment                                                     |
+|--------------|----------|------------|-------------|-------------------------------------------------------------|
+| key          | TEXT     | Hash       | PRIMARY KEY | The `utxo` field, not the hash of the `Enrollment` itself   |
+| val          | BLOB     | Enrollment |             | Binary serialized `Enrollment`, should be converted to text |
+| avail_height | INTEGER  | Height     |             | It's not actually a hash but the key, should be `TEXT`      |
+
+
+## SCP envelope DB
+
+This DB is in its own file and should probably merged in one or the other.
+
+| Field name | SQL Type | D type      | Attributes                | Comment                             |
+|------------|----------|-------------|---------------------------|-------------------------------------|
+| seq        | INTEGER  |             | PRIMARY KEY AUTOINCREMENT | Unused on the D side                |
+| envelope   | BLOB     | SCPEnvelope | NOT NULL                  | Serializing a C++ type is dangerous |
+
+
+## Flash database
+
+This contains Flash-related data.
+
+### `channels` table
+
+| Field name    | SQL Type | D type       | Attributes           | Comment                     |
+|---------------|----------|--------------|----------------------|-----------------------------|
+| chan_id       | BLOB     | Hash         | PRIMARY KEY NOT NULL | Binary-serialized hash      |
+| data          | BLOB     | Transaction  | NOT NULL             | Binary-serialized `Channel` |
+| update_signer | BLOB     | UpdateSigner |                      | Doesn't seem to be used     |
```

Comment:
Is this comment right @AndrejMitrovic ?

Reply:
It's used, just mark it as binary for now.